exchanges/okx/okx_test.go

```diff
@@ -0,0 +1,3169 @@
+
+func TestContractGridAlgoOrdersPushData(t *testing.T) {
+	t.Parallel()
+	if err := ok.WsHandleData([]byte(contractGridAlgoOrdersPushDataJSON)); err != nil {
+		t.Error("Okx Contract Grid Algo Order Push Data error", err)
+	}
+}
+
+var gridPositionsPushDataJSON = `{"arg": {"channel": "grid-positions","uid": "44705892343619584","algoId": "449327675342323712"},"data": [{"adl": "1","algoId": "449327675342323712","avgPx": "29181.4638888888888895","cTime": "1653400065917","ccy": "USDT","imr": "2089.2690000000002","instId": "BTC-USDT-SWAP","instType": "SWAP","last": "29852.7","lever": "5","liqPx": "604.7617536513744","markPx": "29849.7","mgnMode": "cross","mgnRatio": "217.71740878394456","mmr": "41.78538","notionalUsd": "10435.794191550001","pTime": "1653536068723","pos": "35","posSide": "net","uTime": "1653445498682","upl": "232.83263888888962","uplRatio": "0.1139826489932205"}]}`
+
+func TestGridPositionsPushData(t *testing.T) {
+	t.Parallel()
+	if err := ok.WsHandleData([]byte(gridPositionsPushDataJSON)); err != nil {
+		t.Error("Okx Grid Positions Push Data error", err)
+	}
+}
+
+var gridSubOrdersPushDataJSON = `{"arg": {"channel": "grid-sub-orders","uid": "44705892343619584","algoId": "449327675342323712"},"data": [{"accFillSz": "0","algoId": "449327675342323712","algoOrdType": "contract_grid","avgPx": "0","cTime": "1653445498664","ctVal": "0.01","fee": "0","feeCcy": "USDT","groupId": "-1","instId": "BTC-USDT-SWAP","instType": "SWAP","lever": "5","ordId": "449518234142904321","ordType": "limit","pTime": "1653486524502","pnl": "","posSide": "net","px": "28007.2","side": "buy","state": "live","sz": "1","tag":"","tdMode": "cross","uTime": "1653445498674"}]}`
+
+func TestGridSubOrdersPushData(t *testing.T) {
+	t.Parallel()
+	if err := ok.WsHandleData([]byte(gridSubOrdersPushDataJSON)); err != nil {
+		t.Error("Okx Grid Sub orders Push Data error", err)
+	}
+}
+
+func TestGetHistoricTrades(t *testing.T) {
+	t.Parallel()
+	if _, err := ok.GetHistoricTrades(context.Background(), currency.NewPair(currency.BTC, currency.USDT), asset.Spot, time.Time{}, time.Time{}); err != nil {
+		t.Errorf("%s GetHistoricTrades() error %v", ok.Name, err)
+	}
```

Comment:
To demonstrate that the time looping works, please use
```suggestion
	if _, err := ok.GetHistoricTrades(context.Background(), currency.NewPair(currency.BTC, currency.USDT), asset.Spot, time.Now().Add(-time.Hour*2), time.Now().Add(-time.Hour)); err != nil {
		t.Errorf("%s GetHistoricTrades() error %v", ok.Name, err)
	}
```